tree/30_generic_methods/service_action.cf

```diff
@@ -39,7 +39,14 @@ bundle agent service_action(service_name, action)
 
       "action_command"            string => "${paths.path[service]} ${service_name} ${action}";
 
-    !systemctl_utility_present.!service_utility_present.init_d_directory_present::
+    !systemctl_utility_present.!service_utility_present.startsrc_utility_present::
+      "svc_action_cmd[restart]"   string => "/usr/bin/stopsrc -s ${service_name}; /usr/bin/startsrc -s ${service_name}";
```

Comment:
This method may not always work: stopsrc says that the service was requested to stop, not that it stoped
cf https://www.ibm.com/developerworks/community/blogs/brian/entry/scripting_the_stop_and_restart_of_src_controlled_processes_on_aix6?lang=en

the common workaround is:
stopsrc -s sshd
sleep 10
startsrc -s sshd

the tricky one (and hard to make all purpose) is stopsrc -s sshd && until lssrc -s sshd | grep -q inoperative; do echo "Waiting for sshd to stop"; perl -e 'select(undef,undef,undef,.25)'; done && startsrc -s sshd